src/util.js

```diff
@@ -0,0 +1,45 @@
+module.exports = {
+  compact: function(arr) {
+    var compactedArr = [];
+
+    for (var i = 0; i < arr.length; i++) {
+      arr[i] && compactedArr.push(arr[i]);
+    }
+
+    return compactedArr;
+  },
+  clone: function(obj) {
+    return JSON.parse(JSON.stringify(obj));
```

Comment:
This strikes me as such an incredibly fragile `clone` function that it might be dangerous to future generations to label it as a "utility". AFAICT it's only being used in one place--what do you think about pulling it out of utils and including it closer to that one call site?